### PR TITLE
IEN-845 upload bccnm ncas data

### PR DESCRIPTION
**Pull request by sunghwan-park-fw**
- Add endpoints to validate BCCNM/NCAS updates
- Add the table to preview BCCNM/NCAS updates
- Add API and cypress e2e tests

### CODE REVIEW

**Comment by dbayly-freshworks** [comment]
I pulled this down and did some testing, and I found an issue - The column HMBC Unique ID should be matched with the column ats1_id. Its format is a 6 digit integer.

File: .github/workflows/pr-check-e2e.yml [hunk 1]
```diff
@@ -47,3 +47,4 @@ jobs:
           NEXT_PUBLIC_AUTH_REALM: 'ien'
           NEXT_PUBLIC_AUTH_CLIENTID: 'IEN'
           NEXT_PUBLIC_AUTH_URL: 'https://keycloak.freshworks.club/auth'
+          NEXT_PUBLIC_DISABLE_USER_GUIDE: 'true'
```
Comment (dbayly-freshworks) [inline .github/workflows/pr-check-e2e.yml:50]: When running this locally with the user guide disabled, the admin page will get in a loop of requesting the userguide, then displaying a 500 error stating the user guide is disabled, then re-querying for it. It's a bit jarring. If the user guide is disabled, when fetching it, can we instead return an empty array.

Comment (sunghwan-park-fw) [reply .github/workflows/pr-check-e2e.yml:50]: If NEXT_PUBLIC_DISABLE_USER_GUIDE is set, the admin page should not load the user-guide component and there should be no call to fetch user guides. Have you fully refreshed the page after disabling it?

File: packages/common/src/ro/bccnm-ncas-update.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export interface BccnmNcasUpdate {
+  'HMBC Unique ID': string;
+  'Date ROS Contract Signed': string | number;
+  'First Name'?: string;
+  'Last Name'?: string;
+  Email?: string;
```
[the remark this answers is not in the document]
Comment (sunghwan-park-fw) [reply packages/common/src/ro/bccnm-ncas-update.ts:6]: We just create ROS milestones without saving the spreadsheet into the database.
We can add the destination to the ROS milestone's notes field. Let me know if you have any other ideas.

File: apps/api/src/admin/admin.service.ts [hunk 3]
```diff
@@ -104,4 +143,89 @@ export class AdminService {
...
+    if (!applicant) {
+      v.valid = false;
+      v.message = 'Applicant not found';
+      return v;
+    }
+
+    // convert excel date cell value as a number to string
+    const dateOfRosContract = update['Date ROS Contract Signed'];
+    if (typeof dateOfRosContract === 'number') {
+      v.dateOfRosContract = dayjs((+dateOfRosContract - 25568) * 86400 * 1000).format('YYYY-MM-DD');
```
[the remark this answers is not in the document]
Comment (sunghwan-park-fw) [reply apps/api/src/admin/admin.service.ts:165]: I'll add '25568 -> number of days from 1990 to epoch at PST'

**Comment by sunghwan-park-fw** [comment]
> I pulled this down and did some testing, and I found an issue - The column HMBC Unique ID should be matched with the column ats1_id. Its format is a 6 digit integer.

Now all test applicants have ats1_id set and the backend matches applicant.ats1_id to the sheet's `HMBC Unique ID.`

File: packages/common/src/ro/bccnm-ncas-update.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (sunghwan-park-fw) [reply packages/common/src/ro/bccnm-ncas-update.ts:6]: I made the changes in #579

File: scripts/data-applicants.sql
[the remark this answers is not in the document]
Comment (sunghwan-park-fw) [reply scripts/data-applicants.sql:1]: It's a script to generate applicants for testing. I changed `COPY` command  to SQL insert query.

**Pull request merged**